Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -82,72 +100,186 @@ public void OnError(Exception error)
         }
 
         /// <summary>
-        /// Get the DependencyTelemetry objects that are still waiting for a response from the dependency. This will most likely only be used for testing purposes.
+        /// Enables HttpClient instrumentation with activity. It is never called and allows Microsoft.Extensions.DiagnosticAdapter to know that event is enabled.
         /// </summary>
-        internal IEnumerable<DependencyTelemetry> PendingDependencyTelemetry
+        [DiagnosticName("System.Net.Http.HttpRequestOut")]
+        public void IsActivityEnabled()
         {
-            get { return pendingTelemetry.Values; }
         }
 
         /// <summary>
-        /// Diagnostic event handler method for 'System.Net.Http.Request' event.
+        /// Handler for Exception event, it is sent when request processing cause an exception (e.g. because of DNS or network issues)
+        /// There will be anoth Stop event sent with null response.
         /// </summary>
-        [DiagnosticName("System.Net.Http.Request")]
-        public void OnRequest(HttpRequestMessage request, Guid loggingRequestId)
+        [DiagnosticName("System.Net.Http.Exception")]
+        public void OnException(Exception exception, HttpRequestMessage request)
         {
-            try
+            this.client.TrackException(exception);
```

Comment:
eventually. For now - it will be great if you'd extract some infroamtion from the exception like whether it was a DNS resolution error and associate this info with the dependency telemetry object